bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlText.java

```diff
@@ -113,7 +114,7 @@ public boolean hasNext() {
             public Object next() {
                 if (!read) {
                     this.read = true;
-                    return data;
+                    return that;
```

Comment:
Can't we just do 
```suggestion
                    return this;
```

Reply:
Not really sure why but when using `this` the program crashes with a bad sad error for examples like: 

```
xml bitOfText = xml `bit of text\u2702\u2705`;

function xmlCharItemIter() returns string {
    string result = "";
    int i = 0;
    foreach xml elem in bitOfText {
        string str = io:sprintf("%s\n", elem);
        result += str;
        i += 1;
    }
    return result;
}
```

> class io.ballerina.runtime.internal.values.XmlText$1 cannot be cast to class io.ballerina.runtime.internal.values.XmlValue (io.ballerina.runtime.internal.values.XmlText$1 and io.ballerina.runtime.internal.values.XmlValue are in unnamed module of loader 'app')